session1/solutions/assignment4/Rakefile

```diff
@@ -1 +1,46 @@
-# assignment 4 solution
+require 'pathname'
+require 'rubygems/package'
+
+desc 'create archive file from a path'
+task :archive do
+
+  archive_ext = "tar.gz"  
+  path = ENV["PATH2TAR"] || "."
+  exclude = ENV["EXCLUDE"] + "," + archive_ext || archive_ext
+
+  exclude_list = exclude.split(',')
+
+  # check if path exist and set the archive file name
+  if Dir.exist?(path) == false then
+    puts "== archive == Error: Path doesn't exist"
+    exit
+  else
+    Dir.chdir("#{path}")
+    puts "== archive == Archive requested for path: #{Dir.pwd}"
+    tar_name = Pathname.new(Dir.pwd).basename.to_s.gsub(/[^a-z,A-Z]/, '_') + archive_ext
```

Comment:
3 problems here:
1. It was requested to place the file in "build" folder.
2. Final file name is missing a dot (.) before the tar.gz so the name of the archive is <folder-name>tar.gz
3. It looks to me like you doing to much work when the only things you need to to is to extract the folder name from `path` and add `.tar.gz` to it - try to make it simpler as its just not readable